src/lib/lightning/eclair/eclairService.ts

```diff
@@ -250,6 +254,43 @@ class EclairService implements LightningService {
     );
   }
 
+  async addListenerToNode(node: LightningNode): Promise<void> {
+    this.listener = setupListener(this.cast(node));
```

Comment:
There's just a single listener here for all Eclair nodes. If I create and start a network with 3 Eclair nodes, then remove one of the nodes, it will stop listening on all Eclair nodes. We don't want this to happen. We'll probably need to keep a mapping of from node->listener and only close the stop listening on the node that gets removed.

Reply:
Updated